### PR TITLE
[DEST-544][CC-5334][WEBONISE][SEG-4] Add group support to Mixpanel

### PR DESCRIPTION
**Pull request by CarlosMecha**
_This work was originally submitted in #222_

Adds groups support.

### CODE REVIEW

File: integrations/mixpanel/lib/index.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ var Mixpanel = (module.exports = integration('Mixpanel')
   .option('trackAllPages', false)
   .option('trackNamedPages', false)
   .option('trackCategorizedPages', false)
+  .option('groupIdentifierTraits', [])
```
Comment (brennan) [inline integrations/mixpanel/lib/index.js]: Is `groupIdentifierTraits` a new setting? If so, we should should either run a DCS backfill or ensure we don't call any array methods on it until we check that it's not null/undefined and check the type.

File: integrations/mixpanel/lib/index.js [hunk 2]
```diff
@@ -316,6 +317,32 @@ Mixpanel.prototype.alias = function(alias) {
...
+ *
+ * @param {Group} group
+ */
+
+Mixpanel.prototype.group = function(group) {
+  var groupIdentifierTraits = this.options.groupIdentifierTraits;
+  var groupId = group.groupId();
+  var userId = this.analytics.user().id();
+  var traits = group.properties();
+  if (!groupId || !userId || !groupIdentifierTraits.length) {
```
Comment (brennan) [inline integrations/mixpanel/lib/index.js]: Prob should not call `length` until we know whether `groupIdentifierTraits` is not null.

File: integrations/mixpanel/lib/index.js [hunk 3]
```diff
@@ -316,6 +317,32 @@ Mixpanel.prototype.alias = function(alias) {
...
+    return;
+  }
+
+  if (traits && Object.keys(traits).length) {
+    for (var ind = 0; ind < groupIdentifierTraits.length; ind++) {
```
Comment (mericsson) [inline integrations/mixpanel/lib/index.js]: Consistency fix for this PR, let's rename `ind` to `i`.

File: integrations/mixpanel/lib/index.js (hunk 1 above)
Comment (CarlosMecha) [reply integrations/mixpanel/lib/index.js]: Is not a new setting.

File: integrations/mixpanel/lib/index.js (hunk 2 above)
Comment (CarlosMecha) [reply integrations/mixpanel/lib/index.js]: Uhm, good point. I think it's defined by default on the `option` function.

File: integrations/mixpanel/lib/index.js (hunk 3 above)
Comment (CarlosMecha) [reply integrations/mixpanel/lib/index.js]: Sadly we can't. In old versions of Javascript the for loop doesn't have its own scope. If we call it the same way, we'll be reusing the variable. It's a bad practice and the linter also blames me :D

Comment (mericsson) [reply integrations/mixpanel/lib/index.js]: Ooh got it right.

**Pull request merged**